app/controllers/home_controller.rb

```diff
@@ -253,6 +291,27 @@ def stem_resources
     if @lightbox_resource
       @lightbox_resource = materials_data([@lightbox_resource], nil, 4).shift()
       @page_title = @lightbox_resource[:name]
+      page_meta = {
+        :title => nil,
+        :meta_tags => {},
+        :open_graph => {}
+      }
+      page_meta[:title] = @page_title
+      meta_tags = page_meta[:meta_tags]
+      meta_tags[:title] = page_meta[:title]
+      meta_tags[:description] = @lightbox_resource[:description]
+      if meta_tags[:description].blank?
+        meta_tags[:description] = "Check out this educational resource from the Concord Consortium."
+      end
+      open_graph = page_meta[:open_graph]
+      open_graph[:title] = @page_title
+      open_graph[:description] = meta_tags[:description]
+      @resource_icon = @lightbox_resource[:icon]
+      open_graph[:image] = @resource_icon[:url]
+      if open_graph[:image].blank?
+        open_graph[:image] = "https://learn-resources.concord.org/images/stem-resources/stem-resource-finder.jpg"
+      end
+      @open_graph = page_meta[:open_graph]
```

Comment:
Is `page_meta` needed here or do we only need to set `@open_graph`?

This is probably fine but it seems like a bit of duplicated logic. I wonder if it makes sense to have some kind of utility function that handles these #blank? checks. Maybe something like:

```
def open_graph_tags(pairs)
    ret = {}
    pairs.each do |key, value|
        unless value.blank?
            ret[key] = value
        end
    end
    ret
end

```

Then you could call it from the controller methods with:

@open_graph = open_graph_tags(   :title => "foo",
                                :description => nil,
                                :image => "baz" )
```

or

```
@open_graph = open_graph_tags(   :title => nil,
                                :description => nil,
                                :image => ENV['OG_IMAGE_URL'] )
```